CMakeLists.txt

```diff
@@ -1197,9 +1216,15 @@ if(NOT ${universal_build})
         set(ASTCENC_ISA_SSE2 OFF)
     endif()
 
+    if(${ASTCENC_DECOMPRESSOR})
```

Comment:
If somebody sets this and is building the ktx (libktx) target, the build will break due to undefined functions.

Either leave things and they are and always build the full astcenc library, in the knowledge that only called functions will be linked to the shared library or executable, or set up to build both libraries (with the same ISA) and make `ktx` dependent on `astcenc` and `ktx_read` dependent on `astcdec`. The penalty to this approach is that the decompressor will be compiled twice: once for `astcenc` and once for `astcdec`.

Reply:
The root problem I'm trying to fix is that the ASTCENC_DECOMPRESSOR option exists, is shown in `cmake-gui` when configuring this project, sounds like something I'd want to use (as long as there is no way to entirely disable any usage of ASTCENC), but setting it causes non-obvious CMake errors.

Your suggestions wouldn't help with that particular problem

Reply:
I could use `set(ASTCENC_DECOMPRESSOR OFF CACHE INTERNAL "")` to make sure that the ASTCENC_DECOMPRESSOR option is disabled (OFF) and not shown to the user in cmake-gui (INTERNAL)

Reply:
Yes this is the most straightforward thing to do. Setting a local variable with the same name works to hide the configuration variable too. That is how we hide ASTCENC_UNIVERSAL_BUILD.

Sorry I didn't pick up on the root problem before.

Reply:
I mentioned most of it in #996 but I guess I should've mentioned that this option was shown in cmake-gui.
Either way, I replaced my prior fix with `set(ASTCENC_DECOMPRESSOR OFF CACHE INTERNAL "")`